src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+                + " and should be above 18 or equal and less than 115");
+    }
+
+    @Test
+    void register_age_less_than_18_notOk() {
+        userIsValid.setAge(14);
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with age less than 18. "
+                + "Age cannot be a negative number or contains special symbols"
+                + " and should be above 18 or equal and less than 115");
+    }
+
+    @Test
+    void register_age_less_than_0_notOk() {
+        userIsValid.setAge(-19);
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with negative number at his age. "
+                + "Age cannot be a negative number or contains special symbols"
+                + " and should be above 18 or equal and less than 115");
+    }
+
+    @Test
+    void register_user_with_login_that_exist_whitespace_notOk() {
+        userIsValid.setLogin("Good vin");
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with whitespace at his login. "
+                + "Your Login cannot exist whitespace!");
+    }
+
+    @Test
+    void register_login_is_empty_notOk() {
+        userIsValid.setLogin("");
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with empty login. "
+                + "Your Login should exist at least 1 character!");
+    }
+
+    @Test
+    void register_password_less_than_6_characters_notOk() {
+        userIsValid.setPassword("pas");
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with password length less than 6 characters. "
+                + "Your Password should include 6 and more characters!");
+    }
+
+    @Test
+    void register_password_contains_special_characters_notOk() {
+        userIsValid.setPassword("^$#@#($%6*");
+        assertThrows(CustomException.class, () -> {
+            registrationValidator.register(userIsValid);
+        }, "Expected " + CustomException.class.getName()
+                + " to be thrown for the user with password which contains special symbols. "
+                + "Your Password shouldn't contains special characters!");
+    }
+
+    @Test
+    void register_user_with_null_password_notOk() {
```

Comment:
please add test cases for
- null user
- null login
- ok case where everything is ok

Reply:
`null user` case is not addedd

Reply:
But it was added in the first commit (first test - null user case)